.github/workflows/ci.yml

```diff
@@ -18,6 +18,27 @@ jobs:
       - uses: actions/checkout@v3
       - uses: dprint/check@v2.1
 
+  # job to run change detection
+  changes:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      backend: ${{ steps.filter.outputs.backend }}
```

Comment:
🔧 Why only separate between frontend and backend.. we could even be more specific.

e.g. why build the maker-frontend if only the taker-frontend has changed? etc.

I would create an change output for every sub project we have. It might be cumbersome because of dependencies, but I think that's a complexity worth maintaining.

Reply:
Happy to accept further patches! 😁 it will be covered by #3050 (that will cover electron changes, and as a prerequisite for that will have to split maker and taker frontends.)